src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -595,8 +595,14 @@ private static BidRequestCacheInfo bidRequestCacheInfo(ExtRequestTargeting targe
                 final Integer cacheBidsTtl = shouldCacheBids ? cache.getBids().getTtlseconds() : null;
                 final Integer cacheVideoBidsTtl = shouldCacheVideoBids ? cache.getVastxml().getTtlseconds() : null;
 
+                final Boolean returnCreativeBid = cache.getBids() != null ? cache.getBids().getReturnCreative() : null;
```

Comment:
We can use `shouldCacheBids` variable and avoid `cache.getBids() != null` condition:
```
final boolean returnCreativeBid = shouldCacheBids ? ObjectUtils.firstNonNull(cache.getBids().getReturnCreative(), true) : true;
```
Same for `returnCreativeVideoBid`.